Priority Queue/SimplePriorityQueue.cs

```diff
@@ -61,11 +61,11 @@ private SimpleNode GetExistingNode(TItem item)
                 return _nullNodesCache.Count > 0 ? _nullNodesCache[0] : null;
             }
 
-            if (!_itemToNodesCache.ContainsKey(item))
+            IList<SimpleNode> nodes;
+            if (!_itemToNodesCache.TryGetValue(item, out nodes))
```

Comment:
This is actually the one I first spotted that started me looking for what else might be improvable. I had just been looking through to check some fine detail I don't recall, but then noticed the use of contains+get rather than `TryGetValue` and wondered what else there might be.